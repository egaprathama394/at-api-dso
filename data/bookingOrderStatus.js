export default {
  orderDaimoCode: 'DFA202000000123',
  pssOrderStatus: '2',
};
