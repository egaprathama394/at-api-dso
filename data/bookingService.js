const randomstring = require('randomstring');

const fullNameMax = randomstring.generate({
  charset: 'string',
  length: '110',
});

const noPhoneMax = randomstring.generate({
  charset: 'numeric',
  length: '17',
});

const noPhoneRandom = randomstring.generate({
  charset: 'numeric',
  length: '18',
});

const fullNameRandom = randomstring.generate({
  charset: 'string',
  length: '111',
});

const randomNopol1 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol2 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol3 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol4 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

export default {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: '',
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data submit booking service',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: '',
  policeRegNo: `AB${randomNopol1}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for submit booking service max lenght

export const maxInput = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: `${fullNameMax}`,
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data booking service dengan inputan max char',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: `081${noPhoneMax}`,
  policeRegNo: `AC${randomNopol2}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for submit booking service max lenght + 1

export const negativeRequest = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: `${fullNameRandom}`,
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data submit booking service dengan inputan max + 1 char',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: `081${noPhoneRandom}`,
  policeRegNo: `AD${randomNopol3}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for cancel booking service

export const inputDataForCancel = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: '',
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data untuk cancel booking service',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: '',
  policeRegNo: `AA${randomNopol4}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const cancelRequest = {
  orderNo: '',
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

// data for display booking service

export const displayBookingService = {
  isWalkIn: false,
  orderBy: '',
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const invalidDisplayBookingService = {
  isWalkIn: '',
  orderBy: '',
  AccountType: '',
  Action: '',
  ApplicationName: '',
  DeviceType: '',
  OperatingSystem: '',
  UserVersion: '',
};
