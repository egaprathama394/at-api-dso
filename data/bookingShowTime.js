export default {
  branchCode: 'D004',
  scheduleDate: '2020-09-30 00:00:00',
  vehicleType: 'XENIA MT',
  jobs: [
    {
      jobCode: 'OTHER',
      jobDescription: 'PERBAIKAN LAINNYA',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const negativeRequest = {
  branchCode: 'D004',
  scheduleDate: '2018-10-26 00:00:00',
  vehicleType: 'XENIA MT',
  jobs: [
    {
      jobCode: 'OTHER',
      jobDescription: 'PERBAIKAN LAINNYA',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
