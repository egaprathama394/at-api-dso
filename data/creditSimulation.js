export default {
  P_NEW_USED: 'N',
  P_CATEGORY: 'EX2',
  P_VEHICLE_CATEGORY: '1',
  P_BRAND: '002',
  P_TYPE: '1',
  P_MODEL: '1',
  P_DP: '49125000',
  P_OTR: '196500000',
  P_TYPE_INSU: 'A',
  P_MODE_INSU: 'K',
  P_FLAG_ACP: 'Y',
  P_TENORVALUE: '12',
};