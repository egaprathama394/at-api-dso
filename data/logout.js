export default {
  applicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
  deviceID: '8B55A4DA-5331-47F9-9C8D-838A110FA18C',
};


export const negativeRequest = {
  action: 'Display',
  applicationName: '9093D285-5263-4B9A-97D7-C185189B0695',
  userVersion: '1.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
