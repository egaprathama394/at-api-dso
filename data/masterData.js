
export default {
  companyCode: '0003',
  masterDataInfo: [
    {
      lastSync: '1900-01-01 00:00:00',
      objectName: 'Branch',
    },
  ],
  vehicleBrandCode: 'DAIHATSU',
  accountType: '1',
  action: '',
  deviceType: '0',
  operatingSystem: '0',
  serVersion: '1.0.0',
};

export const JobServiceCompany = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'JobService_Company',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const VehicleBrand = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'vehicleBrand',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const district = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'district',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const province = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'province',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const region = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'region',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const holidayCalendar = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'holidayCalendar',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

// looping data for job service company

export const loopVehicleType = [
  {
    vehicleType: 'AYLA',
    title: 'test case get master data job service company where vehicle = AYLA',
  },
  {
    vehicleType: 'GRANMAX',
    title: 'test case get master data job service company where vehicle = GRANMAX',
  },
  {
    vehicleType: 'GRANMAX1.3',
    title: 'test case get master data job service company where vehicle = GRANMAX1.3',
  },
  {
    vehicleType: 'GRANMAX1.5',
    title: 'test case get master data job service company where vehicle = GRANMAX1.5',
  },
  {
    vehicleType: 'LUXIO',
    title: 'test case get master data job service company where vehicle = LUXIO',
  },
  {
    vehicleType: 'Sigra 1.0',
    title: 'test case get master data job service company where vehicle = Sigra 1.0',
  },
  {
    vehicleType: 'Sigra 1.2',
    title: 'test case get master data job service company where vehicle = Sigra 1.2',
  },
  {
    vehicleType: 'SIRION',
    title: 'test case get master data job service company where vehicle = SIRION',
  },
  {
    vehicleType: 'TERIOS',
    title: 'test case get master data job service company where vehicle = TERIOS',
  },
  {
    vehicleType: 'XENIA1.0',
    title: 'test case get master data job service company where vehicle = XENIA1.0',
  },
  {
    vehicleType: 'XENIA1.3',
    title: 'test case get master data job service company where vehicle = XENIA1.3',
  },
  {
    vehicleType: 'OTHERS',
    title: 'test case get master data job service company where vehicle = OTHERS',
  },
];

export const VehicleTestDrive = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'VehicleTestDrive',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const parameterConfigApp = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'ParameterConfigApplication',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
