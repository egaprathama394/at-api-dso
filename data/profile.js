export default {
  action: 'Display',
  applicationName: '9093D285-5263-4B9A-97D7-C185189B0695',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
  deviceID: '74eef5b5-69c0-454d-9e2d-485c9e988766',
};

export const editProfile = {
  userAccount: '',
  AccountType: '1',
  Action: 'Edit',
  ApplicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};
