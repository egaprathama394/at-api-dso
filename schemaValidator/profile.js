export default {
  properties: {
    userAccount: {
      type: 'object',
      properties: {
        email: {
          type: 'string',
        },
        firstName: {
          type: 'string',
        },
        lastName: {
          type: 'string',
        },
        phoneNo: {
          type: 'string',
        },
        address: {
          type: 'string',
        },
        provinceCode: {
          type: 'string',
        },
        districtCode: {
          type: 'string',
        },
        subdistrictCode: {
          type: 'string',
        },
        villageCode: {
          type: 'string',
        },
        zipCode: {
          type: 'string',
        },
        birthdate: {
          type: 'string',
        },
        gender: {
          type: 'integer',
        },
        religion: {
          type: 'integer',
        },
        hobby: {
          type: 'string',
        },
        location: {
          type: 'string',
        },
        identificationType: {
          type: 'string',
        },
        identificationNumber: {
          type: 'string',
        },
        identificationDate: {
          type: 'string',
        },
        urlPhotoProfile: {
          type: 'string',
        },
        occupation: {
          type: 'string',
        },
        defaultPaymentMethod: {
          type: 'integer',
        },
        referralCode: {
          type: 'string',
        },
      },
      required: [
        'email',
        'firstName',
        'lastName',
        'phoneNo',
        'address',
        'provinceCode',
        'districtCode',
        'subdistrictCode',
        'villageCode',
        'zipCode',
        'birthdate',
        'gender',
        'religion',
        'hobby',
        'location',
        'identificationType',
        'identificationNumber',
        'identificationDate',
        'urlPhotoProfile',
        'occupation',
        'defaultPaymentMethod',
        'referralCode',
      ],
    },
    acknowledge: {
      type: 'integer',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
    errorCode: {
      type: 'number',
    },

  },
  required: [
    'userAccount',
    'acknowledge',
    'message',
    'version',
    'build',
  ],
  additionalProperties: false,
};

export const invalidToken = {
  properties: {
    error: {
      type: 'string',
    },
    error_description: {
      type: 'string',
    },
  },
  required: [
    'error',
    'error_description',
  ],
  additionalProperties: false,

};

export const editProfileSchema = {
  properties: {
    acknowledge: {
      type: 'number',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
    errorCode: {
      type: 'number',
    },
  },
  required: [
    'acknowledge',
    'message',
    'version',
    'build',
    'errorCode',
  ],
  additionalProperties: false,
};