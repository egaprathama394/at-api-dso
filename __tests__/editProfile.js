import getProfile from '../endpoints/profile';
import * as profileSchema from '../schemaValidator/profile';
import * as body from '../data/profile';

const fs = require('fs');
const { matchers } = require('jest-json-schema');

expect.extend(matchers);

const ambilToken = fs.readFileSync('./token.json');
const objToken = JSON.parse(ambilToken);

const token = objToken.edit;

describe('Edit Profile', () => {
  test('Can edit profile successfully using authorized token - positive test cases', async () => {
    const response = await getProfile(token, body.default);
    expect(response.status).toEqual(200);
    expect(response.body).toMatchSchema(profileSchema.default);
    body.editProfile.userAccount = await response.body.userAccount;
    body.editProfile.userAccount.firstName = await 'Nama Depan Edit';
    body.editProfile.userAccount.lastName = await 'Nama Belakang Edit';
    body.editProfile.userAccount.phoneNo = await '08100000000';
    body.editProfile.userAccount.birthdate = await '1900-01-01 00:00:00';
    body.editProfile.userAccount.address = await 'Alamat Edit';
    body.editProfile.userAccount.provinceCode = await '3173';
    body.editProfile.userAccount.districtCode = await '3100';
    const responseEdit = await getProfile(token, body.editProfile);
    expect(responseEdit.status).toEqual(200);
    expect(responseEdit.body).toMatchSchema(profileSchema.editProfileSchema);
    expect(responseEdit.body.message).toEqual(`Edit profile for ${body.editProfile.userAccount.email} is success`);
    body.editProfile.userAccount.firstName = await 'Renni';
    body.editProfile.userAccount.lastName = await 'Nugraheni';
    body.editProfile.userAccount.phoneNo = await '081456155090';
    body.editProfile.userAccount.birthdate = await '1993-11-08 00:00:00';
    body.editProfile.userAccount.address = await 'Alamat Asli';
    body.editProfile.userAccount.provinceCode = await '3100';
    body.editProfile.userAccount.districtCode = await '3173';
    const responseReset = await getProfile(token, body.editProfile);
    expect(responseReset.status).toEqual(200);
    expect(responseReset.body).toMatchSchema(profileSchema.editProfileSchema);
    expect(responseReset.body.message).toEqual(`Edit profile for ${body.editProfile.userAccount.email} is success`);
    const responseGet = await getProfile(token, body.default);
    expect(responseGet.status).toEqual(200);
    expect(responseGet.body).toMatchSchema(profileSchema.default);
    Reflect.deleteProperty(responseGet.body, 'build');
    expect(responseGet.body).toMatchSnapshot();
  });
});
