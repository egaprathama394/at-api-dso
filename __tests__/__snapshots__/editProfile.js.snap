// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Profile Can edit profile successfully using authorized token - positive test cases 1`] = `
Object {
  "acknowledge": 1,
  "errorCode": 0,
  "message": "Display profile is success",
  "userAccount": Object {
    "address": "Alamat Asli",
    "birthdate": "1993-11-08 00:00:00",
    "defaultPaymentMethod": 0,
    "districtCode": "3173",
    "email": "testerdso8@gmail.com",
    "firstName": "Renni",
    "gender": 0,
    "hobby": "",
    "identificationDate": "1900-01-01 00:00:00",
    "identificationNumber": "",
    "identificationType": "Other",
    "lastName": "Nugraheni",
    "location": "",
    "occupation": "",
    "phoneNo": "081456155090",
    "provinceCode": "3100",
    "referralCode": "",
    "religion": 0,
    "subdistrictCode": "",
    "urlPhotoProfile": "",
    "villageCode": "",
    "zipCode": "",
  },
  "version": "1.0",
}
`;
